sdk/core/Azure.Core.Experimental/src/MutableJsonElement.ObjectEnumerator.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text.Json;
+
+namespace Azure.Core.Json
+{
+    /// <summary>
+    /// A mutable representation of a JSON element.
+    /// </summary>
+    public readonly partial struct MutableJsonElement
+    {
+        /// <summary>
+        ///   An enumerable and enumerator for the properties of a JSON object.
+        /// </summary>
+        [DebuggerDisplay("{Current,nq}")]
+        public struct ObjectEnumerator : IEnumerable<(string Name, MutableJsonElement Value)>, IEnumerator<(string Name, MutableJsonElement Value)>
```

Comment:
Just curious - why do we use a Tuple in this implementation and a full type in the DynamicJson.ObjectEnumerator?

Reply:
It's because the DLR doesn't handle the field names on Tuples correctly, because the names are just attributes associated with user code.  I have a separate item to make the property Value settable, and some of that work will need to be done in the DynamicJsonProperty Value setter.  If any needs to be added to the tuple setter, we may need to graduate this to a struct, but TBD.